psm_utils/io/idxml.py

```diff
@@ -173,14 +170,16 @@ def _parse_psm(
             score=peptide_hit.getScore(),
             precursor_mz=peptide_id.getMZ(),
             retention_time=peptide_id.getRT(),
-            # NOTE: In future update of OpenMS we will be able to read out ion_mobility from idXML file
+            # NOTE: ion mobility will be supported by OpenMS in the future
             protein_list=[
                 accession.decode() for accession in peptide_hit.extractProteinAccessionsSet()
             ],
             rank=peptide_hit.getRank() + 1,  # 0-based to 1-based
             source="idXML",
-            # Storing proforma notation of peptidoform and UNIMOD peptide sequence for mapping back to original sequence in writer
+            # Storing proforma notation of peptidoform and UNIMOD peptide sequence for mapping back
+            # to original sequence in writer
             provenance_data={str(peptidoform): peptide_hit.getSequence().toString()},
+            # TODO: Can we recover any all UserParams and save them as metadata?
```

Comment:
Yes ill implemented such that all `UserParams` of type `string` will be stored in the metadata dictionary(since that only accepts `{str: str}`. `UserParams` of type `float` are stored in `rescoring_features` *IF* they are in the `RESCORING_FEATURE_LIST`. If you want to open up metadata to metadata: `Optional[Dict[str, Optional[str,float]]` let me know

Reply:
Sounds perfect how you described it!